internal/lockservice/simpleLockService.go

```diff
@@ -187,3 +243,24 @@ func (ls *SimpleLockService) CheckReleased(sd Descriptors) bool {
 	ls.lockMap.Mutex.Unlock()
 	return true
 }
+
+// GetLockMap returns a copy of the current state of the
+// lockservice's lock map.
+// This function is used by the Snapshot() function of the
+// finite state machine of the distributed consensus algorithm
+func (ls *SimpleLockService) GetLockMap() map[string]string {
```

Comment:
You said copy, return the copy.
Here locking is of no use since you're returning the actual map anyway.